pkg/lib/synthesize/generate.go

```diff
@@ -0,0 +1,97 @@
+package synthesize
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+
+	log "github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/runtime"
+	serializer "k8s.io/apimachinery/pkg/runtime/serializer/json"
+)
+
+var CsvInferenceFileNotFoundError = errors.New("a file that contains non-inferable metadata is could not be found")
+
+const SynthesizedCSVFileName string = "olm.clusterserviceversion.yaml"
+
+// Generate generates a CSV for the bundle based on the resource within the bundle directory.
+// The bundleDir points to where the bundle is located, outputDir is where bundle and synthesized CSV will be written.
+// If overwrite is set to true, outputDir is reloaded with the exact content of bundleDir plus a synthesized CSV
+// and inplace generation deletes the CSV if exists.
+func Generate(metadataDir, manifestDirName, outputDir string, overwrite bool) error {
```

Comment:
Are other synthesize functions going to live in this package in the future? If there's even a possibility of that happening, better to rename this to `GenerateCSV` now.